src/gui/search/searchwidget.cpp

```diff
@@ -918,4 +1040,34 @@ void SearchWidget::DataStorage::removeTab(const QString &tabID)
     Utils::Fs::removeFile(makeDataFilePath(tabID + u".json"));
 }
 
+void SearchWidget::DataStorage::loadHistory(const int limit)
+{
+    const Path historyFilePath = makeDataFilePath(HISTORY_FILE_NAME);
+    const auto loadResult = ::loadHistory(historyFilePath, limit);
+    if (!loadResult)
+    {
+        LogMsg(tr("Failed to load Search UI history. File: \"%1\". Error: \"%2\"")
+                .arg(historyFilePath.toString(), loadResult.error()), Log::WARNING);
+        return;
+    }
+
+    emit historyLoaded(loadResult.value());
```

Comment:
Wouldn't it be better that you return parsed `QStringList` here?

Reply:
> Wouldn't it be better that you return parsed `QStringList` here?

Sorry, don't understand what do you mean.

Reply:
Never mind. I missed you already called `::loadHistory()`.